Custom field description

cproensa opened the pull request:
Add description text attribute for custom field definition, and show this description on labels for form inputs.

![cf_description](https://cloud.githubusercontent.com/assets/14123811/14587756/c73b2628-04b8-11e6-8af9-e62dbd6364e8.png)

Fixes #03433

atrol commented:
No time for a deeper look or testing, but looks good at first sight.
What I don't like that much is, that we get one more thing which can't be localised, but that's another story.

Missing documentation, e.g. https://www.mantisbt.org/docs/master/en-US/Admin_Guide/html-desktop/#admin.pages.manage.customfields
https://www.mantisbt.org/docs/master/en-US/Admin_Guide/html-desktop/#admin.customize.customfields.definitions

cproensa commented:
> No time for a deeper look or testing, but looks good at first sight.
What I don't like that much is, that we get one more thing which can't be localised, but that's another story.

i agree. I had opened #0020305 to open discuss possible solutions.

cproensa commented:
> Missing documentation

Actually, cuurent documentation for CF was quite outdated.
I had a revision and clean-up, and added the specific for this PR

vboctor commented on admin/schema.php:
```diff
@@ -865,6 +865,10 @@
 
+# Add description field to custom field definition
+$g_upgrade[206] = array( 'AddColumnSQL', array( db_get_table( 'custom_field' ), "
+	description					XL		NULL DEFAULT NULL" ) );
```
Do we want to support long descriptions here?  Should we use a one line for description?  That would reflect on the db schema and UI.

vboctor replied:
Why not default to empty string rather than null?

vboctor commented on manage_custom_field_edit_page.php:
```diff
@@ -80,6 +80,11 @@
 			</div>
 			<div class="field-container">
+				<label for="custom-field-description"><span><?php echo lang_get( 'description' ) ?></span></label>
+				<span class="textarea"><textarea id="custom-field-description" name="description" cols="70" rows="2"><?php echo string_textarea( $t_definition['description'] ) ?></textarea></span>
```
Single line?

vboctor commented:
@cproensa I haven't revised the documentation, but the rest looks ok.  Thanks for refreshing the documentation.

- single line vs. multi-line? and field time in schema
- null vs empty string in schema

@dregad it would be good to review the schema to make sure no issues there.

cproensa commented:
>    single line vs. multi-line?

I see the use case for multi-line. Usually you wont need more that 2 or 3 lines, but the option is ther if needed.
Notice how i made the textarea 2 rows height, to encourage a short description :)

> field type: long vs varchar vs xxx
>    null vs empty string in schema

I'm neutral on that. I replicated the description field from project table

vboctor commented:
:+1: -- I'll leave it to @dregad to comment on the schema.  I personally prefer empty over null.

dregad commented:
In terms of UI consistency, I'm wondering why the CF's description is sometimes displayed, and sometimes not.

Currently, it's shown when 
- Reporting a new issue
- Changing status to Resolved or Closed

It's missing when
- displaying an issue (view.php)
- updating an issue (bug_update_page.php)
- mass-updating the custom field (bug_actiongroup_page.php)

Haven't had time to review the doc yet.

cproensa commented:
For the missing description:
> displaying an issue (view.php)
> updating an issue (bug_update_page.php)

The table based layout is not easy to accommodate the description in those pages.

> mass-updating the custom field (bug_actiongroup_page.php)

That can be done with the current layout. I will review this.

cproensa commented:
> mass-updating the custom field (bug_actiongroup_page.php)

@dregad, updated the actiongroup page.
and rebased to current head

cproensa commented:
I found the use case for custom fields that may have some restrictions, like min/max lenght, or regexp.
Where it's desirable to explain the valid syntax before the user submits the form (otherwise, getting an error *after* submit, and the risk of losing inputted data).

Using this description to explain the field restrictions can be a solution.

cproensa commented:
> null vs empty string in schema

- changed to default '' instead of null
- rebased to current master-1.3.x
however, this PR is submitted against master (before changing version to 2)
should i resubmit to proper mantis-1.3.x?

sync1 commented:
@vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  whats status ?

l266509 commented:
@vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  Whats status of this PR?

cproensa commented:
outdated & no progress
